src/groups/mqb/mqbstat/mqbstat_queuestats.cpp

```diff
@@ -481,10 +523,91 @@ void QueueStatsDomain::onEvent(EventType::Enum type, bsls::Types::Int64 value)
 void QueueStatsDomain::setQueueContentRaw(bsls::Types::Int64 messages,
                                           bsls::Types::Int64 bytes)
 {
+    BSLS_ASSERT_SAFE(d_statContext_mp && "initialize was not called");
+
     d_statContext_mp->setValue(DomainQueueStats::e_STAT_BYTES, bytes);
     d_statContext_mp->setValue(DomainQueueStats::e_STAT_MESSAGES, messages);
 }
 
+void QueueStatsDomain::reportConfirmTime(bsls::Types::Int64 value,
+                                         const bsl::string& appId)
+{
+    BSLS_ASSERT_SAFE(d_statContext_mp && "initialize was not called");
+
+    // Report `confirm time` metric to the queue context
+    d_statContext_mp->reportValue(DomainQueueStats::e_STAT_CONFIRM_TIME,
+                                  value);
+    if (!d_subContexts_mp) {
+        return;  // RETURN
+    }
+
+    // Report `confirm time` metric to corresponding appId subcontext
+    bsl::list<StatSubContextMp>::iterator it = bsl::find_if(
+        d_subContexts_mp->begin(),
+        d_subContexts_mp->end(),
+        ContextNameMatcher(appId));
+    if (it != d_subContexts_mp->end()) {
+        it->get()->reportValue(DomainQueueStats::e_STAT_CONFIRM_TIME, value);
+    }
+}
+
+void QueueStatsDomain::reportQueueTime(bsls::Types::Int64 value,
+                                       const bsl::string& appId)
+{
+    BSLS_ASSERT_SAFE(d_statContext_mp && "initialize was not called");
+
+    // Report `queue time` metric to the queue context
+    d_statContext_mp->reportValue(DomainQueueStats::e_STAT_QUEUE_TIME, value);
+
+    if (!d_subContexts_mp) {
+        return;  // RETURN
+    }
+
+    // Report `queue time` metric to corresponding appId subcontext
+    bsl::list<StatSubContextMp>::iterator it = bsl::find_if(
+        d_subContexts_mp->begin(),
+        d_subContexts_mp->end(),
+        ContextNameMatcher(appId));
+    if (it != d_subContexts_mp->end()) {
+        it->get()->reportValue(DomainQueueStats::e_STAT_QUEUE_TIME, value);
+    }
+}
+
+void QueueStatsDomain::updateDomainAppIds(
+    const bsl::vector<bsl::string>& appIds)
+{
+    bdlma::LocalSequentialAllocator<2048> localAllocator;
+
+    if (!d_subContexts_mp) {
```

Comment:
Assuming, we do not reconfigure `appIdPostingDomains`

Reply:
Coorect, we only reconfigure domains and their AppIds